src/Resources/Resources/RoleAssignments/NewAzureRoleAssignmentCommand.cs

```diff
@@ -318,14 +318,6 @@ public override void ExecuteCmdlet()
                     Id = ObjectId,
                     ObjectType = ObjectType,
                 },
-                ResourceIdentifier = new ResourceIdentifier()
```

Comment:
I don’t think it is a good choice to remove the whole object “ResourceIdentifier”. It will block NewAzureRoleAssignment other parameters(“ParentResource”, “ResourceGroupName” and so on). 
I think we should check the value of “DefaultProfile” is null? “DefaultProfile.DefaultContext” is null? “DefaultProfile.DefaultContext.Subscription” is null? If we have the value “DefaultProfile.DefaultContext.Subscription.Id” we can assign it to “ResourceIdentifier.Subscription”, else we didn’t set any value to “ResourceIdentifier.Subscription”.

Reply:
The resource identifier object is never used during the flow of NewAzureRoleAssignmentCommand that is why it was removed instead of patched.

Reply:
This `FilterRoleAssignmentsOptions` object is used by [`AuthorizationClient.CreateRoleAssignment`](https://github.com/Azure/azure-powershell/blob/38bb792a85ecd428cee2f58be02018958620898c/src/Resources/Resources/Models.Authorization/AuthorizationClient.cs#L134) and the `ResourceIdentifier` property is never referenced, so I agree that it's OK to be removed.